mathesar/tests/views/api/test_table_api.py

```diff
@@ -1264,3 +1268,136 @@ def test_table_patch_columns_invalid_type_with_multiple_changes(create_data_type
     # The table should not have changed
     original_column_data = _get_data_types_column_data()
     _check_columns(current_table_response.json()['columns'], original_column_data)
+
+
+def test_table_delete_no_primary_key(create_schema, client):
+    table_name = 'NASA Table Delete'
+    schema = create_schema('Patents')
+    engine = create_mathesar_engine(schema.database.name)
+    metadata = MetaData(bind=engine, schema=schema)
+    columns = [
+        Column('id', Integer),
+        Column('name', String)
+    ]
+    table = Table(
+        table_name,
+        metadata,
+        columns,
+        schema=schema
+    )
```

Comment:
You will need to run `sa_table.create()` after this to actually create the table.

Reply:
After this, you'll need to do the following:

`reflect_db_objects()`

This will create the Mathesar model object corresponding to the new table you just created in the database.

`table = Table.objects.get(name=table_name)`

This will get the Mathesar Table object corresponding to the table you've just created. After this, the rest of the test should work fine.

There are multiple types of `Table` objects because we're using a `Table` Django model to represent the underlying SQL Alchemy `Table` object. I know that might be confusing, please let me know if you have further questions about this.

Reply:
I also recommend putting the table creation logic in a new function so that you don't have to repeat it in all the tests.

Reply:
> After this, you'll need to do the following:
> 
> `reflect_db_objects()`
> 

Is this mimicking the behaviour of the `_sa_table` property within the Table model: https://github.com/centerofci/mathesar/blob/839efb5cbf0b9a539490c0a9e125700d4fc2dccb/mathesar/models.py#L150-L162?

Reply:
It's the reverse of it, you're creating the `sa_table` first (which creates the table in the database) and then syncing what's in the database by creating the necessary Django model instance.

You need to do this this way because the default table creation logic in Mathesar will automatically create a primary key field, so you can't use it for these tests.